### PR TITLE
Update .travis.yml

### PR DESCRIPTION
**Pull request by scribam**
- Coverity and Coveralls code analysis are removed as they don't work currently.
- "dist: trusty" is removed as it is now the default Linux distribution (https://blog.travis-ci.com/2017-08-31-trusty-as-default-status)
- Some rework and cleanup added

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,19 +16,10 @@ cache:
   directories:
     - $HOME/hombebrew_cache
 
-env:
-  global:
-    - secure: BXyNziNYFLqLtY6Q58BDEUpRJr81HUtHpxf2Sr646bDcbG0Rf2rb2utqHZJ1Om926WdH7RqHuoVti4dr2X1lh4nHMkLD7MrIrfIkNN5T9aZdk19uw9IyMtz/cXon7/wqVo24tVqtBI0UghFxKii/GTDJEvRKgfpPp1e1vKSNGa4=
```
Comment (AniLeo) [inline .travis.yml]: Don't remove this, I think we're using that, but I'm not entirely sure.

Comment (scribam) [reply .travis.yml]: This line was introduced in commit a792bae1211e299addb7294b56df683e8adba48e which added the support of Coverity Scan. The value has been modified in #1240, #2668 and #2669.
So far, Travis builds fine without the value and it seems unlikely the Appimage upload (which is only done on the master branch) relies on it.
What do you think about it? Do you still want to keep these lines?

Comment (hcorion) [reply .travis.yml]: @AniLeo I think we should just push this, if the AppImage upload no longer works, then do a quick PR reverting that line.

Comment (AniLeo) [reply .travis.yml]: Let's see if this is the secret variable used for AppImage upload then, will merge and check if the script runs

**Pull request merged**